bundles/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/handler/PLCBridgeHandler.java

```diff
@@ -144,62 +160,70 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                            clock = clock.withMonth(buffer[1]);
-                            clock = clock.withDayOfMonth(buffer[2]);
-                            clock = clock.withHour(buffer[3]);
-                            clock = clock.withMinute(buffer[4]);
-                            clock = clock.withSecond(buffer[5]);
-                        } catch (DateTimeException exception) {
-                            clock = ZonedDateTime.now();
-                            logger.info("Return local server time: {}.", exception.getMessage());
+                switch (channelId) {
+                    case RTC_CHANNEL -> {
+                        rtc = ZonedDateTime.now();
+                        if (!LOGO_0BA7.equalsIgnoreCase(getLogoFamily())) {
+                            try {
+                                final var year = rtc.getYear() - rtc.getYear() % 100;
+                                final var date = LocalDate.of(year + buffer[0], buffer[1], buffer[2]);
+                                final var time = LocalTime.of(buffer[3], buffer[4], buffer[5]);
+                                rtc = ZonedDateTime.of(date, time, ZoneId.systemDefault());
+                            } catch (DateTimeException exception) {
+                                logger.info("Return local server time: {}.", exception.getMessage());
+                            }
                         }
+                        updateState(channelUID, new DateTimeType(rtc));
                     }
-                    rtc.set(clock);
-                    updateState(channelUID, new DateTimeType(clock));
-                } else if (DAIGNOSTICS_CHANNEL.equals(channelId)) {
-                    Map<Integer, String> states = LOGO_STATES.get(getLogoFamily());
-                    if (states != null) {
-                        for (Integer key : states.keySet()) {
-                            String message = states.get(buffer[0] & key.intValue());
+                    case DIAGNOSTIC_CHANNEL -> {
+                        final var states = LOGO_STATES.get(getLogoFamily());
+                        if (states != null) {
+                            final var stateText = Integer.toBinaryString(buffer[0]);
+                            var message = translation.getText(bundle, states.get((int) buffer[0]), stateText, null);
+                            for (var bit = 0; bit < Integer.SIZE; ++bit) {
+                                if (((buffer[0] >>> bit) & 1) != 0) {
+                                    message = translation.getText(bundle, states.get(1 << bit), stateText, null);
+                                }
+                            }
                             synchronized (oldValues) {
-                                if (message != null && !Objects.equals(oldValues.get(channelUID), message)) {
+                                if ((message != null) && !Objects.equals(oldValues.get(channelUID), message)) {
```

Comment:
What is the purpose, i find it less to read and unnesesairy
```suggestion
                                if (message != null && !Objects.equals(oldValues.get(channelUID), message)) {
```

Reply:
Thanks: this this will break "received update" rule logic. Removed the all channels state caching.